src/chia_dialect.rs

```diff
@@ -0,0 +1,95 @@
+use std::collections::HashMap;
+
+use crate::allocator::{Allocator, NodePtr};
+use crate::cost::Cost;
+use crate::dialect::Dialect;
+use crate::err_utils::err;
+use crate::f_table::{f_lookup_for_hashmap, FLookup};
+use crate::more_ops::op_unknown;
+use crate::reduction::Response;
+use crate::run_program::OperatorHandler;
+
+const QUOTE_KW: [u8; 1] = [1];
+const APPLY_KW: [u8; 1] = [2];
+
+pub struct OperatorHandlerWithMode {
```

Comment:
this doesn't look like it belongs in the *chia* dialect. The chia dialect is known and doesn't need to be determined at run-time. It doesn't need to keep a run-time lookup-table. In fact, the run-time lookup-table seems like it belongs in the python binding, as it's a trick to allow mixing python functions with rust functions. In the rust API, there's no need for that.